java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/RelationalSymbolicValue.java

```diff
@@ -102,11 +102,18 @@ public RelationalSymbolicValue(Kind kind) {
 
   @Override
   public List<ProgramState> setConstraint(ProgramState initialProgramState, BooleanConstraint booleanConstraint) {
+    return setConstraint(initialProgramState, booleanConstraint, knownRelations(initialProgramState));
+  }
+
+  @Override
+  protected List<ProgramState> setConstraint(ProgramState initialProgramState, Constraint booleanConstraint, Set<RelationalSymbolicValue> knownRelations) {
+    if (!(booleanConstraint instanceof BooleanConstraint)) {
```

Comment:
One option could be to test this against BooleanConstraint.TRUE given that value is tested against FALSE...